src/index.d.ts

```diff
@@ -127,16 +127,16 @@ type InputElement = HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement;
 
 type OwnValue = string | number | boolean | string[];
 
-interface BaseInputProps {
+interface BaseInputProps<T> {
   id: string;
   onChange(event: any): void;
   onBlur(event: any): void;
   value: string;
-  name: string;
+  name: keyof T;
```

Comment:
@danielbuechele  The `StateShape` (here `T`) generic argument is optional - it appears that this is causing problems folks don't pass a state shape generic argument type:


```js
function Form() {
  const [formState, { text }] = useFormState(); // intentionally not passing generic argument
  return (
    <>
      <input {...text('username')} />
       ^^^^^
    </>
  );
}
```

```
Type 'string | number | symbol' is not assignable to type 'DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>'.
  Type 'string' has no properties in common with type 'DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>'.
```

Reply:
Which begs the question, what value do we get out of this change? Using `keyof T` instead of `string`, that is.

All we're doing here is telling React: hey, this input will return a `name` prop of type `string`. It's hard for me to a use case where users are interacting with those props directly to make those types stricter. Can you give me an example of when this will be useful?

Reply:
In my application I have a component like this:

```
interface FormFields {
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
}

function Field({
  formState,
  optional,
  ...props
}: TextInputProps & {
  formState: FormState<FormFields>;
  optional?: true;
  name: keyof FormFields;
}) {
  return (
    <FieldSet
      title={props.title}
      optional={optional}
      error={
        formState.touched[props.name] &&
        formState.validity[props.name] === false &&
        formState.errors[props.name]
      }
    >
      <TextInput {...props} block required={!optional} />
    </FieldSet>
  );
}
```

If `name` is typed as `string`, ` formState.errors[props.name]` will throw an error.

Reply:
Ha! I see!

```
  formState: FormState<FormFields>;
  name: keyof FormFields;
```

Thanks a lot for the example. It's _very_ helpful 😁

Should be fixed in #67